index.js

```diff
@@ -23,64 +33,148 @@ function getAuthStatus(type) {
+  if (!isMac) {
+    return
+  }
+
   if (!['add-only', 'read-write'].includes(accessLevel)) {
     throw new TypeError(`${accessLevel} must be one of either 'add-only' or 'read-write'`)
   }
 
-  return permissions.askForPhotosAccess.call(this, accessLevel)
+  return getPermissionsHandler().askForPhotosAccess.call(this, accessLevel)
 }
 
 function askForInputMonitoringAccess(accessLevel = 'listen') {
+  if (!isMac) {
+    return
+  }
+
   if (!['listen', 'post'].includes(accessLevel)) {
     throw new TypeError(`${accessLevel} must be one of either 'listen' or 'post'`)
   }
 
-  return permissions.askForInputMonitoringAccess.call(this, accessLevel)
+  return getPermissionsHandler().askForInputMonitoringAccess.call(this, accessLevel)
+}
+
+function askForContactsAccess() {
+  if (!isMac) {
+    return
+  }
+
+  return getPermissionsHandler().askForContactsAccess.call(this)
+}
+
+function askForFullDiskAccess() {
+  if (!isMac) {
+    return
+  }
+
+  return getPermissionsHandler().askForFullDiskAccess.call(this)
+}
+
+function askForRemindersAccess() {
+  if (!isMac) {
+    return
+  }
+
+  return getPermissionsHandler().askForRemindersAccess.call(this)
+}
+
+function askForMicrophoneAccess() {
+  if (!isMac) {
+    return
+  }
+
+  return getPermissionsHandler().askForMicrophoneAccess.call(this)
+}
+
+function askForMusicLibraryAccess() {
+  if (!isMac) {
+    return
+  }
+
+  return getPermissionsHandler().askForMusicLibraryAccess.call(this)
+}
+
+function askForSpeechRecognitionAccess() {
+  if (!isMac) {
+    return
+  }
+
+  return getPermissionsHandler().askForSpeechRecognitionAccess.call(this)
 }
 
 module.exports = {
-  askForAccessibilityAccess: permissions.askForAccessibilityAccess,
+  askForAccessibilityAccess: askForAccessibilityAccess,
   askForCalendarAccess: askForCalendarAccess,
-  askForCameraAccess: permissions.askForCameraAccess,
-  askForContactsAccess: permissions.askForContactsAccess,
-  askForFoldersAccess,
-  askForFullDiskAccess: permissions.askForFullDiskAccess,
-  askForInputMonitoringAccess,
-  askForRemindersAccess: permissions.askForRemindersAccess,
-  askForMicrophoneAccess: permissions.askForMicrophoneAccess,
-  askForMusicLibraryAccess: permissions.askForMusicLibraryAccess,
-  askForPhotosAccess,
-  askForSpeechRecognitionAccess: permissions.askForSpeechRecognitionAccess,
-  askForScreenCaptureAccess,
-  getAuthStatus,
+  askForCameraAccess: askForCameraAccess,
+  askForContactsAccess: askForContactsAccess,
+  askForFoldersAccess: askForFoldersAccess,
+  askForFullDiskAccess: askForFullDiskAccess,
+  askForInputMonitoringAccess: askForInputMonitoringAccess,
+  askForRemindersAccess: askForRemindersAccess,
+  askForMicrophoneAccess: askForMicrophoneAccess,
+  askForMusicLibraryAccess: askForMusicLibraryAccess,
+  askForPhotosAccess: askForPhotosAccess,
+  askForSpeechRecognitionAccess: askForSpeechRecognitionAccess,
+  askForScreenCaptureAccess: askForScreenCaptureAccess,
+  getAuthStatus: getAuthStatus,
```

Comment:
FWIW, I think this could be refactored for a much smaller diff:
```
const permissions = process.platform === 'darwin' ? require('bindings')('permissions.node') : undefined

function askForAccessibilityAccess() {
  return permissions?.askForAccessibilityAccess.call(this)
}

function askForCalendarAccess(accessLevel = 'write-only') {
  if (!['write-only', 'full'].includes(accessLevel)) {
    throw new TypeError(`${accessLevel} must be one of either 'write-only' or 'full'`)
  }
  return permissions?.askForCalendarAccess.call(this, accessLevel)
}

odule.exports = {
  askForAccessibilityAccess,
  askForCalendarAccess,
  askForMusicLibraryAccess: permissions?.askForMusicLibraryAccess
  // ...
}
```